proteuslib/tools/parameter_sweep.py

```diff
@@ -15,18 +15,96 @@
 import sys
 import os
 import itertools
+import warnings
 
+from enum import IntEnum, auto
+from abc import abstractmethod, ABC 
 from idaes.core.util import get_solver
 
 # ================================================================
 
+class SamplingType(IntEnum):
+    FIXED = auto()
+    RANDOM = auto()
+
+# ================================================================
+
+class _Sample(ABC): 
+
+    def __init__(self, pyomo_object, *args, **kwargs):
+        if pyomo_object is not None:
```

Comment:
Is there a reason to allow the pyomo_object to be None? If is it None, the current code will crash somewhere...

Reply:
I think I added this to simplify some of the unit tests, e.g., create a sample like `UniformSample(None, range_A[0], range_A[1])` to test only that the random pieces are working.  Admittedly, just passing in a dummy `pyomo_object` isn't much harder and would remove this extra check.